### PR TITLE
198 fix init

### LINKED ISSUE #198: vagrant shell provisioning errors
init_environment.sh has a couple of annoyances and some errors:

I do not know how to resolve these, I will try to install the latest version for ember-cli, but I do not know how it will affect your project.
* npm throws "unmet dependency" warnings about debug when installing ember-cli (upgrading ember-cli could help according to https://github.com/npm/npm/issues/8241)

Example:
```
==> default: npm
==> default:
==> default: WARN
==> default:
==> default: unmet dependency
==> default:  /usr/lib/node_modules/ember-cli/node_modules/express requires debug@'~2.1.1' but will load
```
Full log: https://gist.github.com/1cbfca3c2a8e1677000c.git

* ember throws an error when building
```
==> default: version: 0.1.7
==> default: Could not find watchman, falling back to NodeWatcher for file system events
==> default: Building
==> default: Building...
==> default: Cannot find module 'ember-cli/lib/broccoli/ember-app'
==> default: Error: Cannot find module 'ember-cli/lib/broccoli/ember-app'
==> default:     at Function.Module._resolveFilename (module.js:338:15)
==> default:     at Function.Module._load (module.js:280:25)
==> default:     at Module.require (module.js:364:17)
==> default:     at require (module.js:380:17)
==> default:     at Object.<anonymous> (/var/www/twweb/Brocfile.js:3:16)
==> default:     at Module._compile (module.js:456:26)
==> default:     at Object.Module._extensions..js (module.js:474:10)
==> default:     at Module.load (module.js:356:32)
==> default:     at Function.Module._load (module.js:312:12)
==> default:     at Module.require (module.js:364:17)
==> default:     at require (module.js:380:17)
```

* python managepy run throws error:
```
(twweb)vagrant@precise32:/var/www/twweb$ python manage.py run
 Note: It will take a few seconds for both necessary
      servers to start.  Once you see the message
      message 'Build Successful', both servers are up.

Validating models...

version: 0.1.7
0 errors found
October 08, 2015 - 17:54:29
Django version 1.6.1, using settings 'inthe_am.settings'
Starting development server at http://0.0.0.0:8001/
Quit the server with CONTROL-C.
Could not find watchman, falling back to NodeWatcher for file system events
Cannot find module '/var/www/twweb/node_modules/broccoli-ember-hbs-template-compiler'
Error: Cannot find module '/var/www/twweb/node_modules/broccoli-ember-hbs-template-compiler'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at requireLocal (/usr/lib/node_modules/ember-cli/lib/utilities/require-local.js:6:10)
    at TemplatePlugin.toTree (/usr/lib/node_modules/ember-cli/lib/preprocessors/template-plugin.js:16:10)
    at /usr/lib/node_modules/ember-cli/lib/preprocessors.js:104:26
    at Array.forEach (native)
    at processPlugins (/usr/lib/node_modules/ember-cli/lib/preprocessors.js:103:11)
    at module.exports.preprocessTemplates (/usr/lib/node_modules/ember-cli/lib/preprocessors.js:86:10)
```

* npm throws "No README data" warnings
Example:
```
==> default: npm
==> default:
==> default: WARN
==> default:
==> default: package.json
==> default:  broccoli-asset-rev@2.2.0 No README data
```

Full log: https://gist.github.com/1cbfca3c2a8e1677000c.git

I have attached a patch for these following two annoyances, I did not want to create a pull request for something so small:
* in inthe_am/settings.py line 65, MIDDLEWARE_CLASSES should be list, not tuple
* turn off wget progress bar

Patch: https://gist.github.com/222a0dc6cbfcaed52f28.git

### PR DESCRIPTION
**Pull request by ebayer**
upgraded bower and ember-cli to fix npm and ember related issues, also make some cosmetic changes in vagrant init file

Fixes #198

### CODE REVIEW

File: scripts/vagrant/init_environment.sh [hunk 1]
```diff
@@ -98,33 +108,54 @@ if [ ! -d $TWWEB_TASKD_DATA ]; then
...
+# copy MAIN_DIR into a temp folder and run npm there
+# to avoid EPERM errors on NFS shared folders in vagrant
+# See: https://github.com/npm/npm/issues/3565
+cd $STARTING_DIR
+cp -a $MAIN_DIR /tmp/twweb
+chown -R vagrant. /tmp/twweb
+cd /tmp/twweb
+
+echo "installing ember-cli and bower"
+npm install -g ember-cli bower
```
Comment (coddingtonbear) [inline scripts/vagrant/init_environment.sh]: This step is a little prone to future failure.  As new versions of ember-cli and bower are released, those new versions may be incompatible with libraries we've listed as requirements in `package.json` or `bower.json`, and with this installation command it'll install the latest version available (unless I'm mistaken).

If you have this verifiably working using a specific version of ember-cli and bower, I'd recommend using the `@` syntax (as above) to pin those requirements to the known-working version.

**Comment by coddingtonbear** [comment]
Other than that little thing, this all looks fantastic! I'm going to give this a little closer of a look this afternoon, but I'll definitely be merging this in the next day or so.

Cheers!

File: scripts/vagrant/init_environment.sh (hunk 1 above)
Comment (ebayer) [inline scripts/vagrant/init_environment.sh]: I feel really embarrassed to see that these versions are still correct, npm ls on a successfull build reports those same versions.

I think running the vagrant share on nfs or running ember install before npm install was causing npm package problems from the start.

Comment (coddingtonbear) [inline scripts/vagrant/init_environment.sh]: Aww -- I wouldn't be embarrassed about something like that; that happens to me all the time.  Even without changing the ember-cli/bower versions, I do think that the changes you've made here are valuable.

**Comment by coddingtonbear** [comment]
Thanks, @ebayer, for your contribution here!  Cheers!

**Pull request merged**